### ISSUE TITLE
Feature Proposal : Git integration

### ISSUE THREAD

**Issue by jb-perrier** [FlaxEngine/FlaxEngine#42]
**Feature :**
Git integration to directly keep track of changes in the editor.
Assets would have differents states like Uptodate, Modified, Deleted.
At first the editor would not support the whole git workflow ( just the assets states ), in order to implement other functionnality ( like pull, push , commit, branch ) one after one ( mid-term ).

**Lib to use :**
Since the git integration is only on the editor part, we can use [libgit2sharp](https://github.com/libgit2/libgit2sharp).

Feel free to post your opinion here below.

**Comment by stefnotch** [comment]
I wonder if making this a bit more general would be a good idea. Like, the editor would provide the bare necessities, like interfaces and the ability to show different asset states and then the actual source control integration happens in a well maintained plugin.

**Comment by jb-perrier** [comment]
I agree with you, but i was aiming at something more little ( at least on the beginning ) so it doesn't require much developping time at first.

**Comment by honzapatCZ** [comment]
I dont think its really that good idea, take for example UE4 git integration its utter mess and you loose work all the time. But the editor could have some flag/external tool, to show diffs between 2 files

**Comment by stefnotch** [comment]
Meanwhile the Visual Studio Code git integration is pretty nifty. 
Just because someone else produced a terrible product, it doesn't mean that the Flax-equivalent would also turn out terrible.
